libavcodec/qsv_internal.h

```diff
@@ -24,10 +24,10 @@
 #include "config.h"
 
 #if CONFIG_VAAPI
-#define AVCODEC_QSV_LINUX_SESSION_HANDLE
+#define AVCODEC_QSV_VAAPI_SESSION_HANDLE
```

Comment:
You  should make sure AVCODEC_QSV_LINUX_SESSION_HANDLE (or AVCODEC_QSV_VAAPI_SESSION_HANDLE) is not defined on Windows, otherwise the code below is wrong. 

```
#ifdef AVCODEC_QSV_LINUX_SESSION_HANDLE                                                                                                                                                        
    ret = ff_qsv_set_display_handle(avctx, qs);
    if (ret < 0)
        return ret;
#endif
```
It will try to set display handle even if user specifies d3d11va or dxva2 type.

Reply:
Will change in next push